lib/unleash/scheduled_executor.rb

```diff
@@ -1,19 +1,22 @@
 module Unleash
   class ScheduledExecutor
-    attr_accessor :name, :interval, :max_exceptions, :retry_count, :thread
+    attr_accessor :name, :interval, :max_exceptions, :retry_count, :thread, :immediate_execution
```

Comment:
TBH, I don't like this approach, but we can let it slide for now, if we can refactor in another PR.

Reply:
I have mixed feelings, that has a few impacts I'm not sure about - it means the metrics task loses its ability to have a delayed start and we can only ever eagerly resolve toggles. That's quite a few knock on effects. But I think we should take that to another PR and talk what/why

Reply:
I agree that we should take this in a separate PR.